packages/@sfx/sayt-driver-plugin/test/unit/browser/sayt-driver-plugin.test.ts

```diff
@@ -0,0 +1,193 @@
+      Driver.ready();
+
+      registerListener.getCall(0).calledWith(Driver.saytDataEvent, Driver.fetchSaytData);
+    });
+  });
+
+  describe('.unregister()', () => {
+    let unregisterListener;
+
+    beforeEach(() => {
+      unregisterListener = spy(Driver.core['dom_events'], 'unregisterListener');
+    });
+
+    it('should unregister the sayt event listener', () => {
+      Driver.unregister();
+
+      expect(unregisterListener).to.have.been.calledWith(Driver.saytDataEvent, Driver.fetchSaytData);
+    });
+  });
+
+  describe('.autocompleteCallback', () => {
+    it('should return an array of search term strings from the sayt response', () => {
+      const response = {
+        result: {
+          searchTerms: [
+            { value: 'a' , notValue: 'z' },
+            { value: 'b' },
+            { value: 'c' },
+          ],
+        },
+      };
+
+      const cbReturn = Driver.autocompleteCallback(undefined, response);
+
+      expect(cbReturn).to.deep.equal(['a', 'b', 'c']);
+    });
+
+    it('should only return entries that have a searchterm value', () => {
+      const response = {
+        result: {
+          searchTerms: [
+            { value: 'a' },
+            {},
+            { value: 'c' },
+          ],
+        },
+      };
+
+      const cbReturn = Driver.autocompleteCallback(undefined, response);
+
+      expect(cbReturn).to.deep.equal(['a', 'c']);
+    });
+
+    it('should return an empty array if there are no searchTerms', () => {
+      const response = {
+        result: {
+          searchTerms: [],
+        },
+      };
+
+      const cbReturn = Driver.autocompleteCallback(undefined, response);
+
+      expect(cbReturn).to.deep.equal([]);
+    });
+  });
+
+  describe('.sendSaytAPIRequest()', () => {
+    const saytDataPayload = {
+      query: 'shirt',
+      collection: 'backup',
+    };
+    let autocompleteCallback;
+
+    beforeEach(() => {
+      autocompleteCallback = stub(Driver, 'autocompleteCallback');
+    });
+
+    it('should make a search call through the sayt client', () => {
+      const autocomplete = spy(Driver.core.sayt, 'autocomplete');
+
+      Driver.sendSaytAPIRequest(saytDataPayload);
+
+      expect(autocomplete).to.be.calledWith(
+        saytDataPayload.query,
+        { collection: 'backup' },
+        autocompleteCallback,
+      );
+    });
+
+    it('should return the result of the Sayt API callback', () => {
+      autocompleteCallback.returns(['a', 'b']);
+      const callbackReturn = autocompleteCallback();
+
+      const returnValue = Driver.sendSaytAPIRequest(saytDataPayload);
+
+      expect(returnValue).to.eventually.equal(callbackReturn);
+    });
+  });
+
+  describe('.fetchSaytData()', () => {
+    const query = {
+      query: 'shirt',
+    };
+    let dispatchEvent;
+    let sendSaytAPIRequest;
+
+    beforeEach(() => {
+      sendSaytAPIRequest = stub(Driver, 'sendSaytAPIRequest').callThrough();
```

Comment:
If you're going to `callThrough()`, this may as well be a spy instead. But in either case, it _shouldn't_ call through.

Reply:
Again, if you need the stub to return a promise, you can look at [`stub.resolves`](https://sinonjs.org/releases/latest/stubs/#stubresolvesvalue) and [`stub.rejects`](https://sinonjs.org/releases/latest/stubs/#stubrejects).